src/content/app/genome-browser/state/track-panel/trackPanelSlice.ts

```diff
@@ -284,3 +334,46 @@ export const updateCollapsedTrackIds =
       })
     );
   };
+
+const trackPanelSlice = createSlice({
+  name: 'genome-browser-track-panel',
+  initialState: getTrackPanelState() as TrackPanelState,
+  reducers: {
+    setTrackPanelDataFromUrl(state, action: PayloadAction<ParsedUrlPayload>) {
+      const { activeGenomeId } = action.payload;
+      if (!state[activeGenomeId]) {
+        state.activeGenomeId = getTrackPanelStateForGenome(activeGenomeId);
+      }
+    },
+    updateTrackPanelForGenome(
+      state,
+      action: PayloadAction<{
+        activeGenomeId: string;
+        data: Partial<TrackPanelStateForGenome>;
+      }>
+    ) {
+      const { activeGenomeId, data } = action.payload;
+      const persistentTrackProperties =
+        pickPersistentTrackPanelProperties(data);
+      browserStorageService.updateTrackPanels({
+        [activeGenomeId]: persistentTrackProperties
+      });
```

Comment:
We should keep track of such code and to think how to move it out of reducers. Redux reducers should not perform side effects.

Reply:
I've moved this block out of the reducer here.